src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -1405,6 +1411,33 @@ private RegisteredUserDTO convertUserDOToUserDTO(final RegisteredUser user) {
         return userDTO;
     }
 
+    private List<RegisteredUserDTO> convertUserDOsToUserDTOs(final List<RegisteredUser> users) {
+        List <RegisteredUser> userDOs = users.parallelStream().filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
Objects::nonNull might be such a quick operation that the overheads of using the thread pool makes parallelStream() slow down generating the list rather than speeding it up. Not sure, just a hunch.

Reply:
I don't know what the overheads of threads is, might be worth testing. I suspect you might be right, especially when we don't expect huge lists.